digest.go

```diff
@@ -69,23 +71,37 @@ func (c digestCache) Swap(i, j int) {
 	c[i], c[j] = c[j], c[i]
 }
 
-// Purge removes count oldest entries from DigestAuth.clients
-func (da *DigestAuth) Purge(count int) {
+// Purge removes all clients from DigestAuth.clients, as long as those clients
+// were last seen more than DigestAuth.ClientCacheTTL amount of time ago.
+// If DigestAuth.ClientCacheTTL is unspecified, we will purge 2*DigestAuth.ClientCacheTolerance
+// entries.
+func (da *DigestAuth) Purge() {
 	da.mutex.Lock()
 	defer da.mutex.Unlock()
+
 	entries := make([]digestCacheEntry, 0, len(da.clients))
 	for nonce, client := range da.clients {
+		if da.ClientCacheTTL != DefaultClientCacheTTL {
```

Comment:
[suggestion] Could we eliminate `DefaultClientCacheTTL` and just use zero here instead?

I sort of feel like `DefaultXYZ` should only be used for non-zero values, and that otherwise this can get in the way of readability. In this example here, I'm a little confused why we don't want to do something if the user has configured their `ClientCacheTTL` with some non-default value. If instead we're comparing against zero, then I immediately realize that they just didn't set this field and have a better sense of what's going on here.

Just my opinion though!

Reply:
I had `DefaultClientCacheTTL` specified because I wanted to be explicit that the default was no TTL, but I see your point! Removed the default variable entirely, and am just using `0`.